### PR TITLE
Tests of jump and function tables.

### PR DESCRIPTION
**Pull request by yugr**
Just a bunch of new tests and small refactors.

### CODE REVIEW

File: test/test_regex.py [hunk 1]
```diff
@@ -83,9 +110,41 @@ def test_gdb_stripped_nonpic(self):
...
+        line = '0x0000000000001013 <+19>:	callq  *0x8(%rbx)'
+        call_match = self.strip_regex.match(line)
+
+        self.assertIsNot(call_match, None)
+        self.assertEqual(call_match[1], '19')
+        self.assertEqual(call_match[2], '')
+
+    @unittest.expectedFailure
+    def test_objdump_indirect_call(self):
+        line = '1013:	ff 16                	callq  *(%rsi)'
```
Comment (Kazhuu) [inline test/test_regex.py:131]: Same here. How these should be handled in CFG when we get to that point. Any ideas?

**Review by Kazhuu — APPROVED** [review]
Looks good as always! One note that I could't get the instructions in new test cases from the instruction generation code. Especially this `notrack jmpq`. Also some of the jump instructions do not have end address. I was wondering how we should deal with these when we draw CFG for such function.

File: test/test_regex.py [hunk 2]
```diff
@@ -132,3 +191,23 @@ def test_objdump_conditional(self):
         self.assertIsNot(jump_match, None)
         self.assertEqual(jump_match[1], '100f')
         self.assertEqual(jump_match[2], '0x19')
+
+    @unittest.expectedFailure
+    def test_gdb_jumptable(self):
+        line = '0x000000000000101d <+29>:	notrack jmpq *%rax'
```
[the remark this answers is not in the document]
Comment (yugr) [reply test/test_regex.py:197]: I suspect this depends on the compiler (`notrack` is likely an optimization or something).

File: test/test_regex.py [hunk 3]
```diff
@@ -132,3 +191,23 @@ def test_objdump_conditional(self):
...
+        pattern = asm2cfg.get_jump_pattern(False, 'test_function')
+        jump_match = pattern.search(line)
+
+        self.assertIsNot(jump_match, None)
+        self.assertEqual(jump_match[1], '29')
+        self.assertEqual(jump_match[2], '')
+
+    @unittest.expectedFailure
+    def test_objdump_jumptable(self):
+        line = '101d:	3e ff e0             	notrack jmpq *%rax'
```
[the remark this answers is not in the document]
Comment (yugr) [reply test/test_regex.py:207]: No, it's just a jump to some unknown address which disasm couldn't understand. In that case I think we'll have to add an pointless branch (or leading into special "UNKNOWN" block).

**Pull request merged**